Dockerfiles/Dockerfile.dela

```diff
@@ -1,17 +1,19 @@
 FROM golang:1.20.6-bookworm AS base
 RUN apt-get update && apt-get install git
-# make sure we're using the same head as d-voting
-RUN git clone https://github.com/c4dt/dela.git
-WORKDIR /go/dela/cli/crypto
-RUN go install
 WORKDIR /go/d-voting
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
 COPY . .
-
-FROM base AS build
-COPY --from=base /go/dela .
-COPY --from=base /go/d-voting .
+ENV GOCACHE=/root/.cache/go-build
 WORKDIR /go/d-voting/cli/dvoting
-RUN go build
-ENV PATH=/go/dela/cli/crypto:/go/d-voting/cli/dvoting:${PATH}
+RUN --mount=type=cache,target="/root/.cache/go-build" go install
+# make sure we're using the same head as d-voting
+RUN --mount=type=cache,target="/root/.cache/go-build" cd $( go list -f '{{.Dir}}' go.dedis.ch/dela )/cli/crypto && go install
```

Comment:
isn't that a different DELA repository now?

Reply:
go does some magic here:

```
$ go list -f '{{.Dir}}' go.dedis.ch/dela
.../packages/pkg/mod/github.com/c4dt/dela@v0.0.0-20240125143006-d9bfc5ad2f59
```

and because we kept the `go.dedis.ch/dela` as package name throughout the `github.com/c4dt/d-voting`, and rely on the `replace` directive in `go.mod`, I think it's OK.

_Change my mind_ :)